File: cs50/2019/sentimental/mario-more/mario.py
from cs50 import get_int


def main():
    # Asks for height between 1 and 8
    scale = get_int("Height: ")
    while True:
        if scale > 8 or scale < 1:
            scale = get_int("Height: ")
        break

    # For loop for each line
    for i in range(scale):
        # Calculates left spaces
        space = i - scale + 1
        # Writes the left spaces
        for j in range(-1 * space):
            print(" ", end="")
        # Prints the left hashes
        for k in range(i + 1):
            print("#", end="")
        # Prints the space in the pyramid
        print("  ", end="")
        # Prints the right hashes
        for l in range(i + 1):
            print("#", end="")
        # Prints a new line
        print("")


# Runs the program
if __name__ == '__main__':
    main()

File: cs50/2019/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int get_mario_int(string prompt);

int main(void)
{   
    int height = get_mario_int("Height: ");
    for (int i = height; i > 0; i--)
        {
        int space = i - 1;
            for (int s = space; s > 0; s--) {
                printf(" ");
            }
            for (int x = height - i; x > -1; x--) {
                printf("#");
            }
            printf("  ");
            for (int y = height - i; y > -1; y--) {
                printf("#");
            }
            printf("\n");       
        }  
}

int get_mario_int(string prompt)
{
    int n;
    do
    {
        n = get_int("%s", prompt);
    }
    while (8 < n || n < 1);
    return n;
}